mentat/session_input.py

```diff
@@ -12,8 +12,10 @@
 async def _get_input_request(**kwargs: Any) -> StreamMessage:
     session_context = SESSION_CONTEXT.get()
     stream = session_context.stream
+    default_prompt = session_context.conversation.default_prompt
+    session_context.conversation.default_prompt = ""
```

Comment:
This is sort of a hacky way to pass the information to the front end. I like it passing as the data field but I don't like temporarily storing it in conversation and accessing it here. One alternative would be to have commands return an `Optional[str]` and if they do pass it into `_get_input_request` [here](https://github.com/AbanteAI/mentat/blob/1f21444d9b35caa0a576ad02c38a1f58350afb27/mentat/session_input.py#L47). Btw reading this code I wasn't entirely sure why commands were intercepted there instead of one level up [here](https://github.com/AbanteAI/mentat/blob/1f21444d9b35caa0a576ad02c38a1f58350afb27/mentat/session.py#L118). It's sort of the same thing either way though.

Reply:
I actually do somewhat separate how commands are handled in my agent PR; I made a separate function specifically for intercepting commands

Reply:
I was actually thinking something completely different when it comes to passing up the input from whisper; I was thinking we could just send the input on a completely different channel (so we wouldn't have to touch input_request at all), and the client could have another task listening on that channel that it would use to just add whatever comes in on that channel to prompt_toolkit's buffer. What do you think about that idea? I think it would be cleaner than this and more adaptable for other use cases

Reply:
A different channel and the client storing it makes sense to me. We can't simply add it to the buffer though because it doesn't exist while the command is running. It's made when the input request signal is sent

Reply:
I made a new stream `default_prompt`. I can imagine using it for other things though all the ideas I've come up with so far seem sort of contrived. (Maybe if the user runs `/commit` with no argument gpt could write the commit message and then the user could see `/commit $WHAT_GPT_WROTE` and have a chance to edit before actually commiting?)